cmake/dynamic_analyzer_options.cmake

```diff
@@ -19,7 +19,7 @@ endif()
 
 if(ANALYZE_UNDEFINED)
     # TODO: enable signed-integer-overflow
-    add_compile_options(-fsanitize=undefined -fsanitize=nullability -fno-sanitize=signed-integer-overflow)
+    add_compile_options(-fsanitize=undefined -fsanitize=nullability -fno-sanitize=signed-integer-overflow -fno-sanitize=float-cast-overflow)
```

Comment:
Please provide the UBSAN warning which is being reported here.

Reply:
is `float-cast-overflow` generated from the provided test case ? spontanously that sounds weird to me.

Reply:
From https://clang.llvm.org/docs/UndefinedBehaviorSanitizer.html:

```
-fsanitize=float-cast-overflow: Conversion to, from, or between floating-point types which would overflow the destination. Because the range of representable values for all floating-point types supported by Clang is [-inf, +inf], the only cases detected are conversions from floating point to integer types.
```

Reply:
yeah. So we get a overflow when cppcheck analyze this code:

        check("void f() {\n"
              "    int i = 10;\n"
              "    if(i > 9.9){}\n"
              "    float f = 9.9f;\n"
              "    if(f < 10) {}\n"
              "}\n");

And therefore you have to disable that sanitizer? spontanously I am confused. Shouldn't our analysis be tweaked so the cast overflow is avoided?

Reply:
I do wonder why the data flow thinks that some value is that large/small.. maybe it's some "impossible" value that needs to be casted better?

Reply:
> So we get a overflow when cppcheck analyze this code

We dont get an overflow for that case. We get incorrect rounding. We get the overflow for this case:

```cpp
void foo(double recoveredX) {
  for (double x = 1e-18; x < 1e40; x *= 1.9) {
    double relativeError = (x - recoveredX) / x;
  }
}
```